feat(cli): add json and yaml format in the list sub-command

hanshal101 opened the pull request:
<!-- Thanks for creating this pull request 🤗 Please make sure you followed the conventional commit -->

<!-- If this pull request closes an issue, please mention the issue number below -->
Closes #465 
Fixes #465

## 📑 Description
<!-- Add a brief description of the pr -->
This PR adds the feature of printing the list subcommand in the `json` format.

## ✅ Checks
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required

## ℹ Additional Information
In this code I tried the `JSON.Marshal()` function but the output was not quite readable, hence went with the approach of 
`JSON.MarshalIndent()`.

kosmoz commented on cmd/glasskube/cmd/list.go:
```diff
@@ -19,6 +20,7 @@ type ListCmdOptions struct {
 	ShowDescription   bool
 	ShowLatestVersion bool
 	More              bool
+	ListFormat        string
```
Cobra (the CLI library we use) supports custom types for flags by adding `String()` `Set()` and `Type()` functions to a type. I think it would make sense to use this feature here, since the same flag will be added to multiple commands in the future. 

We do something similar with the bootstrap type, which you can look at for reference: https://github.com/glasskube/glasskube/blob/main/pkg/bootstrap/type.go

Let me know if you have any questions regarding this.

hanshal101 replied:
Well here in the `init` function should have the `VarP` instead of `StringVarP`
Because I took the reference to the Bootstrap Type command and it has `VarP`
https://github.com/glasskube/glasskube/blob/c2d0fcf4306aa9c9f5c716143d644004e6a5f646/cmd/glasskube/cmd/bootstrap.go#L87

kosmoz replied:
That's right. 

First, you add a custom string type. 
```go
type OutputFormat string
```
Then you define possible values.

```go
var (
  OutputFormatJSON OutputFormat = "json"
  OutputFormatYAML OutputFormat = "yaml"
)
```

Next, you add the required functions as mentioned and change the option type and flag declaration as you described.

kosmoz reviewed (changes requested):
Nice work @hanshal101! Do you plan to implement YAML formatting in the same PR? I think it would make sense, then we can close the issue with a single PR.

kosmoz commented on cmd/glasskube/cmd/list.go:
```diff
@@ -109,6 +146,28 @@ func printPackageTable(packages []*list.PackageWithStatus) {
+			fmt.Fprintf(os.Stderr, "error marshaling data to YAML: %v\n", err)
+			cliutils.ExitWithError()
+		}
+		fmt.Println(string(yamlData))
+		fmt.Println("---")
```
This produces an empty document at the end. The boundary marker should be printed before the `yamlData`. Additionally, I think it would make sense to only print the boundary marker if there are at least two packages.

kosmoz reviewed (changes requested):
Works nicely! In addition to the one commend below, I noticed that the output has some mixed casing for the JSON/YAML keys. Can you please add some JSON tags to the `PackageWithStatus` struct at `/pkg/list/list.go`? For example, `Status` should be converted to `status`, so this tag would be appropriate `json:"status,omitempty"`

hanshal101 commented:
Well here I have added a small if else to see that if the current package is the first package or not and moved the `---` above the packages. This will help to eliminate the first package and print in rest others. How's this !

kosmoz commented on cmd/glasskube/cmd/list.go:
```diff
@@ -109,6 +146,36 @@ func printPackageTable(packages []*list.PackageWithStatus) {
+			cliutils.ExitWithError()
+		}
+
+		if isFirstPackage {
+			isFirstPackage = false
+		} else {
+			fmt.Println("---")
+		}
```
I think this could be simplified by utilizing the loop counter instead, like:
```go
for i, pkg := range packages {
    /* ... */
    if i > 0 {
        fmt.Println("---")
    }
    /* ... */
}
```

kosmoz reviewed (changes requested):
Almost perfect! Could you please address these two comments? Afterwards, I think this is ready to merge!
